tests/integration/test_charm.py

```diff
@@ -19,21 +23,80 @@
+)
+PROMETHEUS_K8S = CharmDeploymentConfiguration(
+    entity_url="prometheus-k8s",
+    application_name="prometheus-k8s",
+    channel="latest/edge",
+    trust=True,
+)
+
+
+@pytest.mark.setup
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest, charm_under_test):
+    """Build the charm_under_test and deploy it."""
+    await ops_test.model.deploy(
+        charm_under_test, resources=resources, application_name=APP_NAME, trust=True
+    )
+
+    # Charm will be blocked because it needs prometheus
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="blocked", timeout=1000)
+
+
+@pytest.mark.setup
+@pytest.mark.dependency
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest, this_charm):
-    """Build the charm-under-test and deploy it."""
-    # Deploy the charm and wait for active/idle status
-    await asyncio.gather(
-        ops_test.model.deploy(
-            this_charm, resources=resources, application_name=APP_NAME, trust=True
-        ),
-        ops_test.model.wait_for_idle(
-            apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
-        ),
+async def test_deploy_dependencies(ops_test: OpsTest):
+    """Deploy the integration test dependencies."""
+    await ops_test.model.deploy(**asdict(ISTIO_K8S))
+    await ops_test.model.deploy(**asdict(PROMETHEUS_K8S))
+
+    await ops_test.model.add_relation(ISTIO_K8S.application_name, PROMETHEUS_K8S.application_name)
+
+    await ops_test.model.wait_for_idle(
+        apps=[ISTIO_K8S.application_name, PROMETHEUS_K8S.application_name],
+        status="active",
+        raise_on_blocked=True,
+        timeout=1000,
     )
 
 
+@pytest.mark.setup
+@pytest.mark.abort_on_fail
+async def test_add_relation_prometheus(ops_test: OpsTest):
+    """Relate the charm_under_test to prometheus."""
+    await ops_test.model.add_relation(APP_NAME, PROMETHEUS_K8S.application_name)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=60)
+
+
 @pytest.mark.abort_on_fail
 async def test_kiali_is_available(ops_test: OpsTest):
     """Assert that Kiali is up and available inside the cluster."""
-    assert False, "Test not implemented"
+    # Arrange - get the Kiali service IP
+    kiali_service_ip = get_k8s_service_clusterip(ops_test.model.name, APP_NAME)
+
+    # Assert that Kiali is available via the charm's service
+    assert kiali_service_ip is not None, "Kiali service IP not found"
+    resp = requests.get(url=f"http://{kiali_service_ip}:20001/kiali")
+    assert resp.status_code == 200, "Kiali is not available"
+
+
+@pytest.mark.teardown
+async def test_remove_relation_prometheus(ops_test: OpsTest):
+    """Assert charm is blocked when we remove the prometheus relation."""
+    # pylibjuju's model doesn't have remove_relation (?!)
+    await ops_test.juju("remove-relation", APP_NAME, PROMETHEUS_K8S.application_name)
```

Comment:
```suggestion
await ops_test.model.applications[PROMETHEUS_K8S.application_name].remove_relation(PROMETHEUS_K8S.application_name,APP_NAME)
```
I think this could work?

Reply:
Ah perfect!  ty!  Had to change the suggestion slightly (need a relation name in there) but otherwise perfect!